### PR TITLE
Develop 2016

### PR DESCRIPTION
**Pull request by arthbous**
Changed everything to fenics 2016.

There is still a seg fault in pnp_spheres, I have to try to find where it comes from.

### CODE REVIEW

**Comment by arthbous** [comment]
It works perfectly now. @maximilianmetti 

I suggest the following:
1. Merge develop-2016 to develop 
2. Then delete all the other branches except master and battery-stokes-2016.

File: benchmarks/pnp_spheres/linear_pnp.cpp [hunk 1]
```diff
@@ -100,6 +100,8 @@ dolfin::Function Linear_PNP::fasp_solve () {
     &_amg
   );
 
+  printf("\n### 1.\n");fflush(stdout);
```
Comment (maximilianmetti) [inline benchmarks/pnp_spheres/linear_pnp.cpp]: Can we get rid of this?

File: benchmarks/pnp_spheres/linear_pnp.cpp [hunk 2]
```diff
@@ -108,12 +110,15 @@ dolfin::Function Linear_PNP::fasp_solve () {
...
 
-    dolfin::EigenVector solution_vector(_eigen_vector->size());
+    dolfin::EigenVector solution_vector(_eigen_vector->mpi_comm(),_eigen_vector->size());
     double* array = solution_vector.data();
     for (std::size_t i = 0; i < _fasp_soln.row; ++i) {
+      // printf(" i = %d",i);fflush(stdout);
       array[i] = _fasp_soln.val[i];
     }
 
+  printf("\n### 5.\n");fflush(stdout);
```
Comment (maximilianmetti) [inline benchmarks/pnp_spheres/linear_pnp.cpp]: Can we get rid of this?

File: src/pde.cpp [hunk 3]
```diff
@@ -10,8 +10,8 @@ extern "C" {
   #include "fasp.h"
   #include "fasp_functs.h"
 }
-
-using namespace std;
+// #include <vector>
+// using std::vector;
```
Comment (maximilianmetti) [inline src/pde.cpp]: let's drop these too.

**Review by maximilianmetti — COMMENTED** [review]
Overall, it looks pretty straightforward.  I want to run it on my machine to verify that I get the same performance.  Then we should merge 👍

**Comment by arthbous** [comment]
Yes to all and done

**Comment by arthbous** [comment]
Also I just looked on how you apply EAFE and I like it. Because I realized when we add stokes the DOF changes order it's not 3i, 3i+1, 3i+2, it's totally random.

**Comment by maximilianmetti** [comment]
Yeah, I hope the assumption that the DOFs are ordered is not built in anywhere else in the code... but I think we have avoided it successfully in `develop`.

Sorry that this isn't merged yet.  It's a real job to get FEniCS 2016 working on my machine.  They're requiring _a lot_ of dependencies even for installation...   Hopefully, this will be done soon!

**Pull request merged**